src/rgw/rgw_http_client.cc

```diff
@@ -368,6 +368,53 @@ int RGWHTTPHeadersCollector::receive_header(void * const ptr, const size_t len)
   return 0;
 }
 
+int RGWPostHTTPData::send_data(void* ptr, size_t len)
```

Comment:
This is cleaner.  receive_header is grovelly enough to hide, maybe other refactoring could change that.